src/coreclr/src/vm/weakreferencenative.cpp

```diff
@@ -420,12 +428,17 @@ FCIMPL3(void, WeakReferenceNative::Create, WeakReferenceObject * pThisUNSAFE, Ob
 
     // Create the handle.
 #ifdef FEATURE_COMINTEROP
-    IWeakReference* pRawWinRTWeakReference = GetWinRTWeakReference(&gc.pTarget);
-    if (pRawWinRTWeakReference != nullptr)
+    IWeakReference* pRawComWeakReference = nullptr;
+    if (gc.pTarget != NULL && gc.pTarget->PassiveGetSyncBlock() != nullptr && gc.pTarget->PassiveGetSyncBlock()->GetInteropInfoNoCreate() != nullptr)
```

Comment:
Is the C++ compile smart enough to CSE the redundant call to `pTarget->PassiveGetSyncBlock()` ? I would not count on it.

Reply:
I'll refactor it out.